### PR TITLE
fix(core): use client s3 sdk

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

use client s3 sdk to replace "aws-sdk".

## Testing

Local tested.

## Checklist

- [x] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [ ] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/core/src/utils/storage/s3-storage.ts [hunk 1]
```diff
@@ -8,32 +8,43 @@ export const buildS3Storage = (
   accessKeyId: string,
   secretAccessKey: string
```
Comment (gao-sun) [inline packages/core/src/utils/storage/s3-storage.ts:9]: should we also update this var name for consistency?

Comment (wangsijie) [reply packages/core/src/utils/storage/s3-storage.ts:9]: Actually the name should be `secretAccessKey`, it's better to rename the `accessSecretKey` to `secretAccessKey`, but that is a breaking change, so no hurry to do it.

Comment (gao-sun) [reply packages/core/src/utils/storage/s3-storage.ts:9]: cool. we may keep the backward compatibility and mark deprecated when changing and release all breaking changes when next major release is ready?

Comment (gao-sun) [reply packages/core/src/utils/storage/s3-storage.ts:9]: would be great to have a comment or issue for this though

**Review by gao-sun — APPROVED** [review]

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -4444,13 +5042,40 @@ packages:
...
+    dev: false
+
   /@aws-sdk/middleware-stack@3.224.0:
     resolution: {integrity: sha512-8mBrc3nj4h6FnDWnxbjfFXUPr/7UIAaGAG15D27Z/KNFnMjOqNTtpkbcoh3QQHRLX3PjTuvzT5WCqXmgD2/oiw==}
     engines: {node: '>=14.0.0'}
     dependencies:
       tslib: 2.4.1
     dev: false
 
+  /@aws-sdk/middleware-stack@3.310.0:
```
Comment (gao-sun) [inline pnpm-lock.yaml:5072]: why it added so much dependencies in the lockfile? looks not right

File: packages/core/src/utils/storage/s3-storage.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/utils/storage/s3-storage.ts:9]: LOG-6032

File: pnpm-lock.yaml (hunk 2 above)
Comment (wangsijie) [reply pnpm-lock.yaml:5072]: I've double checked, it has so many dependencies.

**Pull request merged**